stan/math/gpu/opencl_context.hpp

```diff
@@ -194,7 +194,7 @@ class opencl_context {
       // and mark them as compiled.
       for (auto kern : kernel_info()) {
         if (strcmp(kern.second.group, kernel_group) == 0) {
-          kernels()[(kern.first)] = cl::Kernel(program_, kern.first, &err);
+          opencl_context_base::getInstance().kernels[(kern.first)] = cl::Kernel(program_, kern.first, &err);
```

Comment:
I had to change this, as I was getting the "CL_INVALID_KERNEL" when using the kernel (retrieved with get_kernel()). Retrieving the name returned an empty string. I added a test to check this.